### PR TITLE
Cart page

### PR DESCRIPTION
**Pull request by Sudhanshusoni45**
### You will have to login for this to work

### CODE REVIEW

**Comment by Jay-Gadhiya** [comment]
-You can show something like " go to shop" when there is no products in cart.

**Comment by Jay-Gadhiya** [comment]
-In the cart there is a -50 amount is still remaining even if there are no cards, I think you will correct it later.
- In the filter category when I click on the "Jordan" text, it will be clicked on running, which you can fix.

File: src/Context/cart-context.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    cartState.map((item) => {
+      if (item._id === _id) {
+        flag = true;
+      }
+    });
+    return flag;
+  };
+
+  const addToCart = async (product) => {
+    try {
```
Comment (Jay-Gadhiya) [inline src/Context/cart-context.js:44]: you can extract these functions in context for clean code. otherwise it's coded well.

**Review by Jay-Gadhiya — APPROVED** [review]

**Review by RahullRawat — APPROVED** [review]
- Cart functionality is working fine. But there is one edge case i notice that whenever i click multiple times on `qunatity-decrement-btn` , it is going in negative.

Screenshot (160)

- Also in your category filter , whenever i am selecting jordan it is selecting running instead of jordan.
- When user log in , then login button should change to logout button
- After clicking on `Add to cart` btn,  you can simply show a toast that `Added to cart successfully` or you can add a badge on cart icon. This way user will know that Item is added in cart.
- When the cart is empty , you can provide a btn to navigate to products page.

**Pull request merged**